components/ledger/internal/adapters/http/in/transaction_helper.go

```
@@ -0,0 +1,184 @@
+	internalKey string
+}
+
+// buildBalanceOperations constructs and sorts balance operations from the
+// validated transaction entries. This is pure logic with no I/O dependencies.
+// Operations are sorted by internal key to prevent deadlocks in the Lua script.
+//
+// Alias format arriving from the DSL parser: "index#alias#balanceKey"
+// (e.g. "0#@sender#default", "1#@sender#default" for same account appearing twice).
+// SplitAliasWithKey strips the index prefix, returning "alias#balanceKey" for balance lookup.
+func buildBalanceOperations(ctx context.Context, organizationID, ledgerID uuid.UUID, validate *mtransaction.Responses, balances []*mmodel.Balance) []mmodel.BalanceOperation {
+	logger, _, _, _ := libCommons.NewTrackingFromContext(ctx)
```

golangci-lint (2.11.4)

[error] declaration has 3 blank identifiers

(dogsled)